rclpy/src/rclpy/service.hpp

```diff
@@ -17,61 +17,90 @@
+#include "utils.hpp"
+
 namespace py = pybind11;
 
 namespace rclpy
 {
-/// Create a service server
-/**
- * This function will create a service server for the given service name.
- * This service will use the typesupport defined in the service module
- * provided as pysrv_type to send messages over the wire.
- *
- * On a successful call a Capsule pointing to the pointer of the created rcl_service_t *
- * is returned.
- *
- * Raises ValueError if the capsules are not the correct types
- * Raises RCLError if the service could not be created
- *
- * \param[in] pynode Capsule pointing to the node to add the service to
- * \param[in] pysrv_type Service module associated with the service
- * \param[in] service_name Python object for the service name
- * \param[in] pyqos_profile rmw_qos_profile_t object for this service
- * \return capsule containing the rcl_service_t
- */
-py::capsule
-service_create(
-  py::capsule pynode, py::object pysrv_type, std::string service_name,
-  py::object pyqos_profile);
-
-/// Publish a response message
-/**
- * Raises ValueError if the capsules are not the correct types
- * Raises RCLError if the response could not be sent
- *
- * \param[in] pyservice Capsule pointing to the service
- * \param[in] pyresponse reply message to send
- * \param[in] pyheader Capsule pointing to the rmw_request_id_t header of the request we respond to
- */
+
+class Service : public Destroyable
+{
+public:
+  /// Create a service server
+  /**
+   * This class will create a service server for the given service name.
+   * This service will use the typesupport defined in the service module
+   * provided as pysrv_type to send messages over the wire.
+   *
+   * Raises ValueError if the capsules are not the correct types
+   * Raises RCLError if the service could not be created
+   *
+   * \param[in] pynode Capsule pointing to the node to add the service to
+   * \param[in] pysrv_type Service module associated with the service
+   * \param[in] service_name Python object for the service name
+   * \param[in] pyqos_profile QoSProfile Python object for this service
+   * \return capsule containing the rcl_service_t
+   */
+  Service(
+    py::capsule pynode, py::object pysrv_type, std::string service_name,
+    py::object pyqos_profile);
+
+  ~Service() = default;
+
+  /// Publish a response message
+  /**
+   * Raises ValueError if the capsules are not the correct types
+   * Raises RCLError if the response could not be sent
+   *
+   * \param[in] pyresponse reply message to send
+   * \param[in] pyheader Capsule pointing to the rmw_request_id_t header of the request we respond to
+   */
+  void
+  service_send_response(py::object pyresponse, py::capsule pyheader);
+
+  /// Take a request from a given service
+  /**
+   * Raises ValueError if pyservice is not a service capsule
+   * Raises RCLError if the take failed
+   *
+   * \param[in] pyrequest_type Instance of the message type to take
+   * \return None if there was nothing to take, or
```

Comment:
It looks like this PR makes it return `(None, None)` if there was nothing to take